crates/game/src/tree.rs

```diff
@@ -47,12 +42,29 @@ pub fn tree_move_animation(
     let window_width = window.width();
     for (mut transform, _) in tree_query.iter_mut() {
         transform.translation.x = if transform.translation.x < -window_width * 0.8 / 2.0 {
-            status.speed += 2;
+            update_game_speed(&mut status);
             window_width * 0.8 / 2.0
         } else {
-            let more_hard_speed = (status.speed as f32).log10();
+            let more_hard_speed = (status.speed as f32).log2();
             transform.translation.x
                 - time.delta_seconds() * (window_width / 3.0 + (TREE_WIDTH / 2.0) * more_hard_speed)
         };
     }
 }
+
+
+fn update_game_speed(status: &mut GameStatus) {
+    unsafe {
+        if status.speed < MAX_GAME_SPEED {
+            let new_speed = status.speed.saturating_add(SPEED_INCREMENT);
+            SPEED_INCREMENT = SPEED_INCREMENT.saturating_add(SPEED_INCREMENT);
```

Comment:
Speed should not use static mut, consider store it inside component or using bevy resources